tests/shell.shebang.sh

```diff
@@ -1,4 +1,4 @@
 #! @ENV_PROG@ nix-shell
 #! nix-shell -I nixpkgs=shell.nix --no-substitute
-#! nix-shell --pure -i bash -p foo bar
+#! nix-shell  --pure   -i bash -p '(_: foo) "absurd"' "b\ar"  
```

Comment:
This test does not prove enough, because this works:
```
nix-repl> with { foo = true; }; (_: foo) absurd
true
```

What we need to see is that the parsed values are correct, including
 - `"` in `'`
 - `'` in `"`
 - escaping
 - concatenation, for example is this a valid way to handle a single quote in a single-quoted string?
    does `'It'"'"'s one string'` parse as a single string containing `It's one string` ?

To test this effectively, I think it'd be best to find a way to echo a whole string and not just some variables. Maybe `--argstr` could be useful?

Reply:
Done